.github/workflows/codacy.yml

```
@@ -55,17 +55,21 @@ jobs:
-        if: hashFiles('results.sarif') != ''
-        run: ./scripts/merge-sarif-runs.sh results.sarif
+        id: merge-sarif
+        if: steps.codacy-run.conclusion == 'success'
+        run: bash ./scripts/merge-sarif-runs.sh results.sarif
 
-      # Upload the SARIF file (now with single run) using official action
+      # Upload the SARIF file only after successful merge.
+      # Using step outcome instead of hashFiles() ensures we never upload the
+      # raw 23-run file when the merge step fails or is skipped.
       - name: Upload SARIF results file
         uses: github/codeql-action/upload-sarif@89a39a4e59826350b863aa6b6252a07ad50cf83e # v4.32.4
```

YAMLlint (1.38.0)

[warning] too few spaces before comment: expected 2

(comments)